frontend/src/features/permits/pages/Void/FinishVoid.tsx

```diff
@@ -39,8 +42,11 @@ export const FinishVoid = ({
   const { mutation: voidPermitMutation, voidResults } = useVoidPermit();
 
   useEffect(() => {
-    if (voidResults && voidResults.success.length > 0) {
-      // Navigate back to search page
+    const voidFailed = hasPermitsActionFailed(voidResults);
+    if (voidFailed) {
+      onFail();
+    } else if (voidResults && voidResults.success.length > 0) {
```

Comment:
```suggestion
    } else if (voidResults?.success?.length > 0) {
```

Reply:
Out of curiosity, what happens if voidResults?.success?.length is undefined, what happens when undefined > 0? Will it throw an error?

Reply:
undefined > 0 will yield false.

Reply:
Ok I used getDefaultRequiredVal(voidResults?.success?.length) > 0 to check, since typescript was giving errors when doing direct comparison.